### PR TITLE
chore: Refactor Checkbox styles

### PR DESCRIPTION
**Pull request by RayRedGoose**
## Summary

Fixes: #2395 

Refactors styles in `Checkbox` using new style utilities.

## Release Category
Components

### Release Note
`Checkbox` now uses Canvas Tokens and our new styling utilities. The component now supports the `cs` prop, but otherwise the API has not changed. It should behave identically as it did in previous versions.

---

Thank you

### CODE REVIEW

File: modules/react/checkbox/lib/CheckBackground.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        [backgroundVars.outer]: cssVar(brand.alert.darkest, base.cantaloupe400),
+      },
+    },
+  },
+});
+
+export const CheckBackground = createComponent('div')({
+  displayName: 'CheckBackground',
+  Component: ({error, ...elemProps}: any) => {
+    return <div {...mergeStyles(elemProps, checkboxBackgroundStyles({error}))} />;
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckBackground.tsx:42]:
```suggestion
    return <Element {...mergeStyles(elemProps, checkboxBackgroundStyles({error}))} />;
```

File: modules/react/checkbox/lib/CheckBackground.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        [backgroundVars.inner]: cssVar(brand.alert.base, base.cantaloupe600),
+        [backgroundVars.outer]: cssVar(brand.alert.darkest, base.cantaloupe400),
+      },
+    },
+  },
+});
+
+export const CheckBackground = createComponent('div')({
+  displayName: 'CheckBackground',
+  Component: ({error, ...elemProps}: any) => {
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckBackground.tsx:41]:
```suggestion
  Component: ({error, ...elemProps}: any, Element) => {
```

Comment (mannycarrera4) [reply modules/react/checkbox/lib/CheckBackground.tsx:41]: Not sure how  we want to handle this, do we use `Box as={Element}` and then for the props you can say `ExtractProps` instead of having an `any` here?

Comment (mannycarrera4) [reply modules/react/checkbox/lib/CheckBackground.tsx:41]: or maybe just ```{error, ...elemProps}: CSProp``` ?

File: modules/react/checkbox/lib/CheckboxCheck.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        '& > div': {
+          backgroundColor: brand.primary.base,
+        },
+      },
+    },
+  },
+});
+
+const indeterminateBoxStyles = createStyles({
+  width: calc.add(system.space.x2, px2rem(2)),
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckboxCheck.tsx:54]: `add` is also exported yeah? idk i like the object notation

File: modules/react/checkbox/lib/CheckboxRipple.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  height: calc.add(system.space.x4, px2rem(2)),
+  width: calc.add(system.space.x4, px2rem(2)),
+  transition: 'box-shadow 150ms ease-out',
+  position: 'absolute',
+  pointerEvents: 'none',
+});
+
+export const CheckboxRipple = createComponent('span')({
+  displayName: 'CheckboxRipple',
+  Component: elemProps => {
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckboxRipple.tsx:18]:
```suggestion
  Component: (elemProps, Element) => {
```

File: modules/react/checkbox/lib/CheckboxRipple.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    return <span className={checkboxRippleStyles} {...elemProps} />;
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckboxRipple.tsx:19]:
```suggestion
    return <Element className={checkboxRippleStyles} {...elemProps} />;
```

Comment (mannycarrera4) [reply modules/react/checkbox/lib/CheckboxRipple.tsx:19]: do we want to use `handleCSProp` here?

File: modules/preview-react/form-field/lib/FormFieldLabel.tsx [hunk 6]
```diff
@@ -4,11 +4,11 @@ import {createSubcomponent} from '@workday/canvas-kit-react/common';
 import {type} from '@workday/canvas-kit-react/tokens';
 import {Text} from '@workday/canvas-kit-react/text';
 import {useFormFieldLabel, useFormFieldModel} from './hooks';
-import {createStencil, createStyles} from '@workday/canvas-kit-styling';
+import {CSProps, createStencil, createStyles} from '@workday/canvas-kit-styling';
 import {mergeStyles} from '@workday/canvas-kit-react/layout';
 import {base, brand, system} from '@workday/canvas-tokens-web';
 
-export interface FormFieldLabelProps {
```
Comment (alanbsmith) [inline modules/preview-react/form-field/lib/FormFieldLabel.tsx:11]: Question: Double-checking. Is this related to this PR? Looks like it's not used in Checkbox?

File: modules/react/checkbox/lib/CheckboxInput.tsx [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+          borderColor: backgroundVars.inner,
+          boxShadow: `0 0 0 1px ${backgroundVars.inner}, 0 0 0 2px ${backgroundVars.outer}`,
+        },
+        '&:where(:checked, :indeterminate) ~ div:first-of-type': {
+          borderColor: 'transparent',
+          boxShadow: `
+          0 0 0 2px ${base.frenchVanilla100},
+          0 0 0 4px ${backgroundVars.inner},
+          0 0 0 5px ${backgroundVars.outer}`,
+        },
```
Comment (alanbsmith) [inline modules/react/checkbox/lib/CheckboxInput.tsx:201]: Question (not RC): I wonder if some of these shared styles should be consolidated?

File: modules/react/checkbox/lib/CheckboxInput.tsx [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+        '&:where(:hover, .hover) ~ span:first-of-type': {
+          boxShadow: 'none',
+        },
+      },
+    },
+    error: {
+      error: {
+        '&:not(:where(:focus-visible, .focus)) ~ div:first-of-type': {
+          borderColor: backgroundVars.inner,
+          boxShadow: `0 0 0 1px ${backgroundVars.inner}, 0 0 0 2px ${backgroundVars.outer}`,
```
Comment (alanbsmith) [inline modules/react/checkbox/lib/CheckboxInput.tsx:193]:
```suggestion
          boxShadow: `0 0 0 ${px2rem(1)} ${backgroundVars.inner}, 0 0 0 ${px2rem(2)} ${backgroundVars.outer}`,
```

File: modules/react/checkbox/lib/CheckboxInput.tsx (hunk 7 above)
Comment (alanbsmith) [reply modules/react/checkbox/lib/CheckboxInput.tsx:201]:
```suggestion
          boxShadow: `
          0 0 0 ${px2rem(2)} ${base.frenchVanilla100},
          0 0 0 ${px2rem(4)} ${backgroundVars.inner},
          0 0 0 ${px2rem(5)} ${backgroundVars.outer}`,
        },
```

File: modules/react/checkbox/lib/CheckboxInput.tsx [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+        },
+        '&:not(:where(:checked, :indeterminate, :disabled, :focus-visible, .focus)):where(:hover, .hover, :active, .active) ~ div:first-of-type':
+          {
+            borderColor: backgroundVars.inner,
+          },
+      },
+      alert: {
+        '&:not(:where(:focus-visible, .focus)) ~ div:first-of-type': {
+          border: `1px solid ${backgroundVars.inner}`,
+          boxShadow: `0 0 0 1px ${backgroundVars.inner}, 0 0 0 2px ${backgroundVars.outer}`,
```
Comment (alanbsmith) [inline modules/react/checkbox/lib/CheckboxInput.tsx:210]:
```suggestion
          border: `${px2rem(1)} solid ${backgroundVars.inner}`,
          boxShadow: `0 0 0 ${px2rem(1)} ${backgroundVars.inner}, 0 0 0 ${px2rem(2)} ${backgroundVars.outer}`,
```

File: modules/react/checkbox/lib/CheckboxInput.tsx [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+        '&:not(where(:checked, :indeterminate, :disabled, :focus-visible, .focus)):where(:hover, .hover, :active, .active) ~ div:first-of-type':
+          {
+            borderColor: backgroundVars.inner,
+          },
+        '&:where(:checked, :indeterminate) ~ div:first-of-type': {
+          borderColor: 'transparent',
+          boxShadow: `
+                0 0 0 2px ${base.frenchVanilla100},
+                0 0 0 4px ${backgroundVars.inner},
+                0 0 0 5px ${backgroundVars.outer}`,
```
Comment (alanbsmith) [inline modules/react/checkbox/lib/CheckboxInput.tsx:221]:
```suggestion
                0 0 0 ${px2rem(2)} ${base.frenchVanilla100},
                0 0 0 ${px2rem(4)} ${backgroundVars.inner},
                0 0 0 ${px2rem(5)} ${backgroundVars.outer}`,
```

File: modules/react/checkbox/lib/CheckboxInput.tsx [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+        },
+      },
+    },
+  },
+  compound: [
+    {
+      modifiers: {variant: 'inverse', error: 'error'},
+      styles: {
+        '&:not(:where(:focus-visible, .focus)) ~ div:first-of-type': {
+          border: `1px solid ${base.soap300}`,
```
Comment (alanbsmith) [inline modules/react/checkbox/lib/CheckboxInput.tsx:231]:
```suggestion
          border: `${px2rem(1)} solid ${base.soap300}`,
```

File: modules/react/checkbox/lib/CheckboxInput.tsx [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+        '&:where(:checked, :indeterminate) ~ div:first-of-type': {
+          borderColor: base.soap300,
+        },
+      },
+    },
+    {
+      modifiers: {variant: 'inverse', error: 'alert'},
+      styles: {
+        '&:not(:where(:focus-visible, .focus)) ~ div:first-of-type': {
+          border: `1px solid ${base.soap300}`,
```
Comment (alanbsmith) [inline modules/react/checkbox/lib/CheckboxInput.tsx:246]:
```suggestion
          border: `${px2rem(1)} solid ${base.soap300}`,
```

File: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx [hunk 13]
```diff
@@ -22,7 +22,8 @@ any questions.
   - [Card](#card)
   - [Count Badge](#count-badge)
   - [Text](#text)
-	- [Form Field Preview](#form-field-preview)
+  - [Form Field Preview](#form-field-preview)
+  - [Checkbox](#checkbox)
```
Comment (alanbsmith) [inline modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:26]: Nit: Let's keep this list alphabetized, please. 🙏🏼

File: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx [hunk 14]
```diff
@@ -142,298 +143,198 @@ instead.
...
-			<Table.Header scope="col">v11 Value</Table.Header>
-		</Table.Row>
+    <Table.Row>
+      <Table.Header scope="col">Component</Table.Header>
+      <Table.Header scope="col">Variant</Table.Header>
+      <Table.Header scope="col">psuedo Class</Table.Header>
+      <Table.Header scope="col">Property</Table.Header>
+      <Table.Header scope="col">v10 Value</Table.Header>
+      <Table.Header scope="col">v11 Value</Table.Header>
+    </Table.Row>
```
Comment (alanbsmith) [inline modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:165]: Thanks for fixing these. 🙏🏼

**Review by alanbsmith — CHANGES_REQUESTED** [review]
Nice work, @RayRedGoose! I made some comments and suggestions.

- Let's go ahead and convert all `px` values to `rem`
- Be sure to check styles like `marginLeft` to make sure they still work with RTL

Otherwise, this looks great. 😄

File: modules/react/checkbox/lib/CheckBackground.tsx [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+    boxSizing: 'border-box',
+    display: 'flex',
+    height: calc.add(system.space.x4, px2rem(2)),
+    justifyContent: 'center',
+    padding: `${system.space.zero} ${calc.divide(system.space.x1, 2)}`,
+    pointerEvents: 'none',
+    position: 'absolute',
+    transition: 'border 200ms ease, background 200ms',
+    width: calc.add(system.space.x4, px2rem(2)),
+    border: `1px solid ${base.licorice200}`,
```
[the remark this answers is not in the document]
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckBackground.tsx:23]: you want to go wild. okay, i understand this :)

File: modules/react/checkbox/lib/CheckBackground.tsx (hunk 1 above)
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckBackground.tsx:42]: Why? There is no way to reassign it. It's not exported by consumer component

File: modules/react/checkbox/lib/CheckBackground.tsx (hunk 2 above)
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckBackground.tsx:41]: It's utility component for `Checkbox`, not separately usable component

File: modules/react/checkbox/lib/CheckboxCheck.tsx (hunk 3 above)
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckboxCheck.tsx:54]: yes, Nicholas said use it with `calc` to help transformer understand it

File: modules/preview-react/form-field/lib/FormFieldLabel.tsx (hunk 6 above)
Comment (RayRedGoose) [reply modules/preview-react/form-field/lib/FormFieldLabel.tsx]: removed

File: modules/react/checkbox/lib/CheckboxRipple.tsx [hunk 16]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import {createComponent} from '@workday/canvas-kit-react/common';
+import {calc, createStyles, px2rem} from '@workday/canvas-kit-styling';
+import {base, system} from '@workday/canvas-tokens-web';
+
+const checkboxRippleStyles = createStyles({
+  borderRadius: system.shape.round,
+  boxShadow: `0 0 0 0 ${base.soap200}`,
```
[the remark this answers is not in the document]
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckboxRipple.tsx]: Okay, I changed it to 'none' and it made some changes to inverse variant:

old
Screenshot 2024-02-06 at 11 10 14 AM

new
Screenshot 2024-02-06 at 11 10 23 AM

As you can see old version has small round inside box

[the remark this answers is not in the document]
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckboxRipple.tsx]: I don't think this round should be visible at all

[the remark this answers is not in the document]
Comment (RayRedGoose) [reply modules/react/checkbox/lib/CheckboxRipple.tsx]: If you agree to remove it, please, approve visual changes in chromatic

File: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx [hunk 17]
```diff
@@ -347,6 +348,14 @@ See below:
 and [tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs). `Card`
 no longer extends the `Box` component, however, it still supports Emotion `styled` and style props.
 
+### Checkbox
+
+**PR:** [#2542](https://github.com/Workday/canvas-kit/pull/2542)
+
+`Checkbox` now uses Canvas Tokens and our new styling utilities. The component now supports the `cs`
```
Comment (mannycarrera4) [inline modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:355]:
```suggestion
`Checkbox` now uses [Canvas Tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs) and our new styling utilities](https://workday.github.io/canvas-kit/?path=/docs/styling-basics--create-modifiers#createstyles-api). The component now supports the `cs`
```

File: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx [hunk 18]
```diff
@@ -347,6 +348,14 @@ See below:
 and [tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs). `Card`
 no longer extends the `Box` component, however, it still supports Emotion `styled` and style props.
 
+### Checkbox
+
+**PR:** [#2542](https://github.com/Workday/canvas-kit/pull/2542)
+
+`Checkbox` now uses [Canvas Tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs) and our new styling utilities](https://workday.github.io/canvas-kit/?path=/docs/styling-basics--create-modifiers#createstyles-api). The component now supports the `cs`
```
Comment (mannycarrera4) [inline modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:355]:
```suggestion
`Checkbox` now uses [Canvas Tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs) and our [new styling utilities](https://workday.github.io/canvas-kit/?path=/docs/styling-basics--create-modifiers#createstyles-api). The component now supports the `cs`
```

Comment (mannycarrera4) [reply modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:355]: sorry

File: modules/react/checkbox/lib/CheckBackground.tsx [hunk 19]
```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import {createComponent} from '@workday/canvas-kit-react/common';
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckBackground.tsx:2]:
```suggestion
import {createComponent, ErrorType} from '@workday/canvas-kit-react/common';
```

File: modules/react/checkbox/lib/CheckboxContainer.tsx [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+  Component: ({children, disabled, id, label, variant}: CheckboxContainerProps) => {
+    return (
+      <div className={checkboxContainerStyles}>
+        <div>{children}</div>
+        {label && (
+          <LabelText
+            htmlFor={id}
+            disabled={disabled}
+            variant={variant}
+            cs={{paddingInlineStart: cssVar(system.space.x3), cursor: 'pointer'}}
```
Comment (mannycarrera4) [inline modules/react/checkbox/lib/CheckboxContainer.tsx:47]:
```suggestion
            cs={{paddingInlineStart: cssVar(system.space.x3), cursor: disabled ? 'default' : 'pointer'}}
```

**Review by mannycarrera4 — APPROVED** [review]
✅ 🥊

**Review by alanbsmith — APPROVED** [review]
